layer/contracts/logged_data.py

```diff
@@ -14,6 +14,140 @@
     import torch
 
 
+class Video:
+    def __init__(
+        self,
+        video: "torch.Tensor",
+        fps: Union[float, int] = 4,
+    ):
+        """
+        :param video: Supported video types are:
+        - torch.Tensor (tensor shape must be NTCHW)
+        :param fps: Frames per second
+        """
+        self.video = video
+        self.fps = fps
+
+    def get_video(self) -> Path:
+        if "torch" in get_base_module_list(self.video):
+            try:
+                import torch
+
+                assert isinstance(self.video, torch.Tensor)
+
+                return Video._convert_to_video_path(self.video, self.fps)
+            except ImportError:
+                raise Exception(
+                    "You need torch & torchvision installed to log torch.Tensor videos. Install with: `pip install torch torchvision`"
+                )
+        else:
+            raise Exception(
+                "Unsupported video type! Supported video types are: torch.Tensor"
+            )
+
+    # Inspired from https://github.com/pytorch/pytorch/blob/ed0091f8db1265449f13e2bdd1647bf873bd1fea/torch/utils/tensorboard/summary.py#L507
```

Comment:
I think we need to make attribution on the NOTICE file that we have code derived from there.

Reply:
Something like:
```
The class layer.contracts.logged_data.Video contains code originating
from the pytorch project (https://github.com/pytorch/pytorch/blob/ed0091f8db1265449f13e2bdd1647bf873bd1fea/torch/utils/tensorboard/summary.py#L507).
```

Reply:
I'll put something to the NOTICE file. Perhaps I don't need anything else? https://discuss.pytorch.org/t/licence-for-a-wrapper/82453/5

Reply:
Better to put the LICENSE header of the pytorch in the class contains code from there.